src/contract.rs

```diff
@@ -45,6 +53,26 @@ pub fn instantiate(
 
     TOKEN2.save(deps.storage, &token2)?;
 
+    let owner = msg.owner.map(|h| deps.api.addr_validate(&h)).transpose()?;
+    OWNER.save(deps.storage, &owner)?;
+
+    let protocol_fee_recipient = deps.api.addr_validate(&msg.protocol_fee_recipient)?;
+    let total_fee_percent = msg.lp_fee_percent + msg.protocol_fee_percent;
+    let max_fee_percent = Decimal::from_str(MAX_FEE_PERCENT)?;
```

Comment:
Doing the const time stuff above will mean that we can avoid this function which looks pretty gnarly: https://docs.rs/cosmwasm-std/latest/src/cosmwasm_std/math/decimal.rs.html#298-336